sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -190,6 +190,13 @@ private[hive] class HiveClientImpl(
       Hive.set(clientLoader.cachedHive.asInstanceOf[Hive])
     }
     SessionState.start(state)
+    // SPARK-21067: Initialize the HdfsEncryptionShim and the underlying FileSystem(FS) eagerly.
+    // The underlying FS is cached and shared among sessions if the sessions request for FS with
+    // the same schema, authority and ugi. If we don't do this, the initialization of the FS will
+    // be delayed to when one session call `getHdfsEncryptionShim` explicitly. However the FS is
+    // auto-closable, which would be closed when the session is closed. The subsequent sessions use
+    // the same FS will throw "Filesystem closed Exception".
+    state.getHdfsEncryptionShim
```

Comment:
If we set `fs.hdfs.impl.disable.cache=true` in Hadoop conf ([PR#25058](https://github.com/apache/spark/pull/25058)), I think results are the same. FileSystem will be created when we get a new session.
So maybe as @jerryshao commented, it's unnecessary to do code changes in Spark?

Reply:
Is this the right place for this? HDFS can be used without `hive` module.

Reply:
@Deegue , if `fs.hdfs.impl.disable.cache=true`, this issue(SPARK-21067) would never happen, and this code has nothing effect.
Pls correct me if i'm wrong.

Reply:
@dongjoon-hyun , ehmm, I think it's the "right" place. The issue is because spark hive module doesn't use the HdfsEncryptionShim **carefully**. As for other modules, that's another thing.